core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java

```diff
@@ -102,4 +103,25 @@ private InternalNumericMetricsAggregation(String name, List<PipelineAggregator>
     protected InternalNumericMetricsAggregation(StreamInput in) throws IOException {
         super(in);
     }
+
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(format, innerHashCode());
+    }
+
+    // norelease: make this abstract when all InternalAggregations implement this method
+    protected int innerHashCode() {
+        return System.identityHashCode(this);
+    }
+
+    @Override
+    protected boolean doEquals(Object obj) {
+        InternalNumericMetricsAggregation other = (InternalNumericMetricsAggregation) obj;
+        return Objects.equals(format, other.format) && innerEquals(other);
+    }
+
+    // norelease: make this abstract when all InternalAggregations implement this method
+    protected boolean innerEquals(InternalNumericMetricsAggregation obj) {
+        return this == obj;
+    }
```

Comment:
I don't like having both doEquals and innerEquals. Maybe just override equals to also include the format and still expect sub classes to implement doEquals? (Same for the hashcode)